### PR TITLE
AM::Delivery job configuration

### PR DESCRIPTION
**Pull request by tamird**
@matthewd @senny

### CODE REVIEW

**Comment by matthewd** [comment]
To me, the whole point of having ApplicationJob is to allow external code to safely inherit from AJ::Job, and trust it will work normally.

That breaks down if our default recommendation is to leave AJ::Job without any queue configured, and only set it for ApplicationJob.

(Actually... AM::DeliveryJob doesn't inherit from ApplicationJob, does it? I certainly don't *think* it should.)

**Comment by tamird** [comment]
@matthewd good points. I've removed the second commit and changed the title of the PR.

File: guides/source/active_job_basics.md [hunk 1]
```diff
@@ -269,6 +269,12 @@ UserMailer.welcome(@user).deliver_now
 UserMailer.welcome(@user).deliver_later
 ```
 
+You can also configure `ActionMailer::DeliveryJob` to use a specific queue
+adapter, distinct from the one set on `ActiveJob::Base`:
```
Comment (matthewd) [inline guides/source/active_job_basics.md]: Isn't this lying? The code seems to be explicitly setting it to `:inline`, and never deferring to the one set on AJ::Base.

(FWIW, I prefer the behaviour described in the comment over [my understanding of] the behaviour in the railtie.)

File: guides/source/active_job_basics.md [hunk 2]
```diff
@@ -269,6 +269,12 @@ UserMailer.welcome(@user).deliver_now
...
+
+```ruby
+Rails.application.config.action_mailer.delivery_job.queue_adapter = :sneakers
```
Comment (matthewd) [inline guides/source/active_job_basics.md]: The two-level name, and user-visible mention of `delivery_job`, seems unfortunate. My first inclination is to call it `action_mailer.delivery_queue` or similar... but maybe we do have other configurable parameters on the job, that people would want to set?

File: guides/source/active_job_basics.md (hunk 1 above)
Comment (tamird) [inline guides/source/active_job_basics.md]: Oh, I see what you mean - I'll remove `options.delivery_job.queue_adapter ||= :inline`. Nice catch!

File: guides/source/active_job_basics.md (hunk 2 above)
Comment (tamird) [inline guides/source/active_job_basics.md]: Yeah, my thinking was that folks might want to manipulate other parameters down the road.

**Comment by tamird** [comment]
PTAL

**Comment by tamird** [comment]
Ping.

**Comment by matthewd** [comment]
I think this is good to go, given a changelog entry... though I do wonder if @jeremy's suggestion in https://github.com/rails/rails/issues/19793#issuecomment-163343022 could somehow be applied here... essentially making the mailer feel progressively more like it *is* the job -- even if, due to technical details, it's not actually the thing that lives on a queue.

**Comment by senny** [comment]
@matthewd this looks like a competing idea to #22270. I think switching out the class itself will open more flexibility. What do you think?

**Comment by tamird** [comment]
Added changelog entry.

**Comment by tamird** [comment]
@senny looks like #22270 was closed. I've rebased this, PTAL

**Comment by senny** [comment]
@tamird isn't configuring the queue_adapter something that you could easily do by reopening the `DeliveryJob` class? Then the same argument for closing #22270 would apply to this change as well.

**Comment by tamird** [comment]
@senny yeah, I suppose you could. If that's the suggested solution, go ahead and close this.

**Comment by senny** [comment]
Yes opening reopening the `DeliveryJob` class is the advised way to go.
Closing for the reasons outlined in https://github.com/rails/rails/pull/22270#issuecomment-166980892

@tamird thanks for your work.

**Pull request closed without merging**